packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -158,6 +296,13 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
     public getChild(id: string): ISummarizerNodeWithGC | undefined {
         return this.children.get(id) as SummarizerNodeWithGC;
     }
+
+    /**
+     * Override the hasChanged method and add a condition to check if this node's used state changed.
+     */
+    protected hasChanged(): boolean {
+        return this.referenceUsedState !== this.used || super.hasChanged();
```

Comment:
I wonder why we have to special case this.referenceUsedState. I.e. we also generate (and store) a blob with used routes, and if that changed, node is changed.
I.e. other way to say it, should it be more generic than picking up boolean from a blob?

Reply:
That's a great point. It will also help with the case where this node's used state did not change but one or more of its sub-node's used state changed. In this case, we need to update this node's summary as well.

Currently, we are not writing the used routes to the summary because I thought writing a boolean is enough. But it might be better to write the routes.